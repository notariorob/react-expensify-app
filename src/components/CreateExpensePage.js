import React from 'react';

const CreateExpensePage = () => (
  <div>
    This is from my add expense component
  </div>
);

export default CreateExpensePage;