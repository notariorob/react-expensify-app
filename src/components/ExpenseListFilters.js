import React from 'react';
import { connect } from 'react-redux';
import { DateRangePicker } from 'react-dates';
import 'react-dates/initialize';
import 'react-dates/lib/css/_datepicker.css';
import { setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate } from '../actions/filters';

class ExpenseListFilters extends React.Component {
  state = {
    calendarFocused: null
  }

  onDatesChange = ({ startDate, endDate }) => {
    this.props.dispatch(setStartDate(startDate));
    this.props.dispatch(setEndDate(endDate));
  }

  onFocusChange = (calendarFocused) => {
    this.setState(() => ({ calendarFocused }));
  }

  render() {
    return (
      <div>
        <input 
          type="text" 
          value={this.props.filters.text} 
          onChange={(e) => {
            this.props.dispatch(setTextFilter(e.target.value));
          }}
        />
        <select 
          onChange={(e) => {
            if (e.target.value === 'date') {
              this.props.dispatch(sortByDate());
            } else if (e.target.value === 'amount') {
              this.props.dispatch(sortByAmount());
            }
          }}
        >
          <option value="date">Date</option>
          <option value="amount">Amount</option>
        </select>
        <DateRangePicker
          numberOfMonths={1}
          isOutsideRange={() => false}
          showClearDates={true}
          startDateId=""
          endDateId=""
          startDate={this.props.filters.startDate}
          endDate={this.props.filters.endDate}
          onDatesChange={this.onDatesChange}
          focusedInput={this.state.calendarFocused}
          onFocusChange={this.onFocusChange}
        />
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    filters: state.filters 
  }
};

const ConnectedExpenseListFilters = connect(mapStateToProps)(ExpenseListFilters);

export default ConnectedExpenseListFilters;