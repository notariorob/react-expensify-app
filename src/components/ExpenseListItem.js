import React from 'react';
import { connect } from 'react-redux';

import { removeExpense, editExpense } from '../actions/expenses';

 const ExpenseListItem = ({ dispatch, id, description, amount, createdAt }) => (
  <div>
    <h2>{description}</h2>
    <p>Amount: {amount}</p>
    <p>Created at: {createdAt}</p>
    <button onClick={() => { 
      dispatch(removeExpense({id}));
    }}>Remove</button>
  </div>
);

const ConnectedExpenseListItem = connect()(ExpenseListItem);

export default ConnectedExpenseListItem;