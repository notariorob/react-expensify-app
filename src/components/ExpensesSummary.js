import React from 'react';
import numeral from 'numeral';
import { connect } from 'react-redux';

import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expensesTotal';

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
  const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
  const formattedExpenseTotal = numeral(expensesTotal / 100).format('$0,00.00');

  return (
    <div>
      <h2>Viewing {expenseCount} {expenseWord} totalling {formattedExpenseTotal}</h2>
    </div>
  )
}

const mapStateToProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters)
  return {
    expenseCount: visibleExpenses.length,
    expensesTotal: selectExpensesTotal(visibleExpenses)
  }
}

export default connect(mapStateToProps)(ExpensesSummary);
