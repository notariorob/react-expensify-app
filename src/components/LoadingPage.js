import React from 'react';

const LoadingPage =  () => (
  <div className="loader">
    <img src="/images/loader.gif" alt="Loading spinner" className="loader__image"/>
  </div>
);

export default LoadingPage;