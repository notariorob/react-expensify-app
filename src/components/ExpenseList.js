import React from 'react';
import { connect } from 'react-redux';

import selectExpenses from '../selectors/expenses';
import ExpenseListItem from './ExpenseListItem';

export const ExpenseList = (props) => (
  <div className="content-container">
    <div className="list-header">
      <div className="mobile-only">Expenses</div>
      <div className="desktop-only">Expense</div>
      <div className="desktop-only">Amount</div>
    </div>
    <div class="list-body">
      {
        props.expenses.length === 0 ? (
          <div className="list-item list-item--message">
            <span>No expenses</span>
          </div>
        ) : (
          props.expenses.map((expense) => {
            return <ExpenseListItem key={expense.id} {...expense} />
        })
        )
      }    
    </div>
  </div>
);

const mapStateToProps = (state) => {
  return {
    expenses: selectExpenses(state.expenses, state.filters)
  };
};


export default connect(mapStateToProps)(ExpenseList);