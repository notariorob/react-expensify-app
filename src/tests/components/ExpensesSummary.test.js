import React from 'react';
import { shallow } from 'enzyme';

import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should render ExpensesSummary correctly when 1 expense is provided', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={235} />);

  expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary correctly when various expenses are provided', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={5} expensesTotal={123456789} />);

  expect(wrapper).toMatchSnapshot();
});