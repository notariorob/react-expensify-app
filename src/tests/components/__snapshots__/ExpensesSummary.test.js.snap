// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary correctly when 1 expense is provided 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
       totalling 
      <span>
        $2.35
      </span>
    </h1>
    <div
      className="page-header__action"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should render ExpensesSummary correctly when various expenses are provided 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        5
      </span>
       
      expenses
       totalling 
      <span>
        $1,234,567.89
      </span>
    </h1>
    <div
      className="page-header__action"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
